src/logger.ts

```diff
@@ -0,0 +1,135 @@
+import { format } from 'util';
+
+// safely preserve unpatched console.* methods in case of compat require
+const unpatchedConsole = {
+  debug: console.debug,
+  info: console.info,
+  log: console.log,
+  warn: console.warn,
+  error: console.error,
+};
+
+// Determine if structured logs are supported (node >= 10). If something goes wrong,
+// assume no since unstructured is safer.
+const SUPPORTS_STRUCTURED_LOGS =
+  parseInt(process.versions?.node?.split('.')?.[0] || '8', 10) >= 10;
+
+// Map LogSeverity types to their equivalent `console.*` method.
```

Comment:
If this goes to public docs, let's backtick `LogSeverity` or just say "log severity" in the text flow.

Reply:
Won't be in public docs.